app/src/components/modals/ActivityBottomSheet/FilterActivity.jsx

```diff
@@ -0,0 +1,141 @@
+import { CategoryContext } from '../../../store/contexts/categoryContext';
+import filterButtonTypes from '../../../docs/filterMarkers';
+import colors from '../../../../colors';
+import { ActivitiesContext } from '../../../store/contexts/activitiesContext';
+
+const filterTitle = (title, icon) => (
+    <View className="flex-row space-x-2 items-center">
+        <Icon name={icon} color={colors.primary.DEFAULT} />
+        <Text className="font-ms-medium text-base">{title}</Text>
+    </View>
+);
+
+const ViewWithDivider = ({ children }) => (
+    <View className="border-b border-b-gray-200 py-4">{children}</View>
+);
+
+export const ActivitiesFilterBottomSheet = ({
+    handleCloseModal,
+    storedActivities,
+    storedCategories,
+    storeFilterSelection,
+}) => {
+    const bottomSheetRef = useRef(null);
+    const { categories } = useContext(CategoryContext);
+    const { getActivityList } = useContext(ActivitiesContext);
+    const { height } = Dimensions.get('window');
+    const isBigPhone = height > 720;
+    const [isLoading, setIsLoading] = useState(false);
+    const [inputedActivities, setInputedActivities] =
+        useState(storedActivities);
+    const [inputedCategories, setInputedCategories] =
+        useState(storedCategories);
+
+    const removeFromState = (id, list, setter) => {
+        const removeId = list.filter((idFromState) => idFromState !== id);
+        setter(removeId);
+    };
+
+    const chipsSelection = (id, list, setter) => {
+        if (list.includes(id)) removeFromState(id, list, setter);
+        else setter([...list, id]);
+    };
+
+    const updateActivitiesList = async () => {
+        const categories = inputedCategories.length ? inputedCategories : null;
+        const activities = inputedActivities.length ? inputedActivities : null;
+        await getActivityList(categories, activities);
+        setIsLoading(false);
+    };
+
+    const filterButtonAction = async () => {
+        setIsLoading(true);
+
+        setTimeout(() => {
+            storeFilterSelection(inputedActivities, inputedCategories);
+            (async () => {
+                await updateActivitiesList();
+                bottomSheetRef.current.dismiss();
+            })();
+        }, 0);
+    };
+
+    const mapChips = (list, type) => {
+        const activityArray =
+            type === 'categorias' ? inputedCategories : inputedActivities;
+        return (
+            <View className="flex-row flex-wrap">
+                {list.map((activity) => {
+                    const isDisabled =
+                        type === 'categorias' &&
+                        inputedCategories.length >= 3 &&
+                        !inputedCategories.includes(activity._id);
+
+                    const selected = activityArray.includes(activity._id);
+
+                    return (
+                        <Chips
+                            title={activity.name}
+                            key={activity._id + activityArray}
```

Comment:
Opa, será que aqui é necessário passar todo um array como key? Será que não podemos apenas utilizar o id + o type?
```suggestion
                            key={activity._id + type}
```

Reply:
feito